File: edukate-backend/src/main/java/io/github/sanyavertolet/edukate/backend/repositories/FileObjectRepository.java
package io.github.sanyavertolet.edukate.backend.repositories;

import io.github.sanyavertolet.edukate.backend.entities.files.FileObject;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@Repository
public interface FileObjectRepository extends ReactiveMongoRepository<FileObject, String> {

    Mono<FileObject> findByKeyPath(String keyPath);

    Flux<FileObject> findAllByKeyPathStartingWith(String prefix);

    Mono<Long> deleteByKeyPath(String keyPath);
}

File: edukate-backend/src/main/kotlin/io/github/sanyavertolet/edukate/backend/repositories/FileObjectRepository.kt
package io.github.sanyavertolet.edukate.backend.repositories

import io.github.sanyavertolet.edukate.backend.entities.files.FileObject
import org.springframework.data.repository.reactive.ReactiveCrudRepository
import org.springframework.stereotype.Repository
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

@Repository
interface FileObjectRepository : ReactiveCrudRepository<FileObject, Long> {
    fun findByKeyPath(keyPath: String): Mono<FileObject>

    fun findAllByKeyPathStartingWith(prefix: String): Flux<FileObject>

    fun deleteByKeyPath(keyPath: String): Mono<Long>
}
